linux_os/guide/system/logging/rsyslog_sending_messages/rsyslog_remote_tls/ansible/shared.yml

```
@@ -5,21 +5,87 @@
-  shell: sed -e '/^action\s*(\s*type\s*=\s*"omfwd"/,/)/!d' /etc/rsyslog.conf /etc/rsyslog.d/*.conf || true
-  register: include_omfwd_config_output
-
-- name: "Get include files directives"
-  shell: >
-    set -o pipefail
-    echo \"{{ include_omfwd_config_output.stdout }}\"|grep  'StreamDriver=\"gtls\"'
-  register: include_omfwd_gtls_config_output
-  when: (include_omfwd_config_output.stdout_lines| length > 0)
-
-- name: "Set rsyslog omfwd to use TLS"
-  lineinfile:
+- name: "{{{ rule_title }}}: search for omfwd action directive in rsyslog include files"
+  ansible.builtin.find:
+    paths: "/etc/rsyslog.d/"
+    pattern: "*.conf"
+    contains: '^\s*action\s*\(\s*type\s*=\s*"omfwd".*'
+  register: rsyslog_includes_with_directive
+
+- name: "{{{ rule_title }}}: search for omfwd action directive in rsyslog main config file"
+  ansible.builtin.find:
+    paths: "/etc"
+    pattern: "rsyslog.conf"
+    contains: '^\s*action\s*\(\s*type\s*=\s*"omfwd".*'
+  register: rsyslog_main_file_with_directive
+
+- name: "{{{ rule_title }}}: declare Rsyslog option parameters to be inserted inf entirely missing"
+  ansible.builtin.set_fact:
+    rsyslog_parameters_to_add_if_missing:
+      - "protocol"
+      - "target"
+      - "port"
+      - "StreamDriver"
+      - "StreamDriverMode"
+      - "StreamDriverAuthMode"
+      - "streamdriver.CheckExtendedKeyPurpose"
+
+- name: "{{{ rule_title }}}: declare Rsyslog option values to be inserted if entirely missing"
+  ansible.builtin.set_fact:
+    rsyslog_values_to_add_if_missing:
+      - "tcp"
+      - "{{ rsyslog_remote_loghost_address }}"
+      - "6514"
+      - "gtls"
+      - "1"
+      - "x509/name"
+      - "on"
+
+- name: "{{{ rule_title }}}: declare Rsyslog option parameters to be replaced if defined with wrong values"
+  ansible.builtin.set_fact:
+    rsyslog_parameters_to_replace_if_wrong_value:
+      - "protocol"
+      - "StreamDriver"
+      - "StreamDriverMode"
+      - "StreamDriverAuthMode"
+      - "streamdriver.CheckExtendedKeyPurpose"
+
+- name: "{{{ rule_title }}}: declare Rsyslog option values to be replaced when having wrong value"
+  ansible.builtin.set_fact:
+    rsyslog_values_to_replace_if_wrong_value:
+      - "tcp"
+      - "gtls"
+      - "1"
+      - "x509/name"
+      - "on"
+
+- name: "{{{ rule_title }}}: assemble list of files with existing directives"
+  ansible.builtin.set_fact:
+    rsyslog_files: "{{ rsyslog_includes_with_directive.files | map(attribute='path') | list + rsyslog_main_file_with_directive.files | map(attribute='path') | list }}"
+
+- name: "{{{ rule_title }}}: try to fix existing directives"
+  block:
+    - name: "{{{ rule_title }}}: Fix existing omfwd directives by adjusting the value"
+      ansible.builtin.replace:
+        path: "{{ item[0] }}"
+        regexp: '(?i)^(\s*action\s*\(\s*type\s*=\s*"omfwd"[\s\S]*)({{ item[1][0] | regex_escape() }}\s*=\s*"\S*")([\s\S]*\))$'
+        replace: '\1 {{ item[1][0] }}="{{ item[1][1] }}"\3'
```

```suggestion
        replace: '\1{{ item[1][0] }}="{{ item[1][1] }}"\3'
```
This extra space between the reference and parameter should be removed to ensure idempotence. Otherwise, each time the task is executed it will include an extra space and report as `changed`.